soccer/MainWindow.cpp

```diff
@@ -878,24 +877,81 @@ void MainWindow::on_actionStopRobots_triggered() {
     SimCommand cmd;
     // TODO: check that this handles threads properly
     for (OurRobot* robot : state()->self) {
-        SimCommand::Robot* r = cmd.add_robots();
-        r->set_shell(robot->shell());
-        r->set_blue_team(processor()->blueTeam());
-        r->mutable_vel()->set_x(0);
-        r->mutable_vel()->set_y(0);
-        r->set_w(0);
+        if (robot->visible) {
+            SimCommand::Robot* r = cmd.add_robots();
+            r->set_shell(robot->shell());
+            r->set_blue_team(processor()->blueTeam());
+            Geometry2d::Point newPos =
+                _ui.fieldView->getTeamToWorld() * robot->pos;
+            r->mutable_pos()->set_x(newPos.x);
```

Comment:
For whatever reason SimCommand doesn't work if the position is not set. It just prints "Trying to override non-existent robot".